plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sqs.ts

```diff
@@ -67,9 +68,13 @@ export class SqsServiceExtension implements ServiceExtension {
           spanAttributes[SemanticAttributes.MESSAGING_OPERATION] =
             MessagingOperationValues.RECEIVE;
 
-          request.commandInput.MessageAttributeNames = (
-            request.commandInput.MessageAttributeNames ?? []
-          ).concat(propagation.fields());
+          const messageAttributeNames = request.commandInput.MessageAttributeNames ?
+          deduplicateMessageAttributeNames(request.commandInput.MessageAttributeNames) : [];
+
+          request.commandInput.MessageAttributeNames = [
+            ...messageAttributeNames,
+            ...propagation.fields()
```

Comment:
I think that this array that we build can now contain the `propagation` fields more than once. What do you think about calling `deduplicateMessageAttributeNames` after the arrays are merged?

Reply:
The concern we have about calling `deduplicateMessageAttributeNames` after merging the arrays is that we are then calling deduplicate every time, which it seemed like we were trying to avoid based on your earlier comment.

Reply:
Sorry, I think I wasn't clear before
What I had in mind is something like:
```js
request.commandInput.MessageAttributeNames = 
    request.commandInput.MessageAttributeNames?.length
            ? deduplicateMessageAttributeNames(
                [...request.commandInput.MessageAttributeNames, ...propagation.fields()]
              )
           : propagation.fields()
```

If I'm not mistaken, it will call `deduplicateMessageAttributeNames` only once, only if the `MessageAttributeNames` are set to something, and after merging them with `propagation.fields()` (which guarantees that the final array being sent to SQS server will not contain duplicates)

In the common case, `MessageAttributeNames` are not set and it will fallback to the `:` part and store `propagation.fields()` directly without any intermediate arrays.

This code can be prettified a bit by extracting some of the logic into function/s, but that is a matter of personal preference :) 

What do you think?

Reply:
👋🏽 I've been working with @JamieDanielson on this fix and we're happy to make this change. The one issue we see with this is that in one case we are de-duping `propagation.fields()` and in the other case we aren't. Would it make more sense to make sure they are either consistent across both cases? So either we don't merge the arrays while de-duping or we don't have the check so it's always de-duped? This also may be a misunderstanding in case we always expect `propagation.fields()` to have unique values in which case this change makes sense! Let us know what you think 🙂

Reply:
Hi @pkanal , thank you for working on this fix.

I think the assumption is that `propagation.fields()` are expected to be distinct.
Fields can be duplicated when we merge `propagation.fields()` into the request`MessageAttributeNames` which can already contain them, so to my understanding we only need to deduplicate when merging. If you want to be extra defensive and deduplicate all the time that is also fine with me :)

But no matter what we decide, I suggest that the final array being sent to SQS server will not contain duplicates by design, which is not currently guaranteed as the arrays are merged **after** calling `deduplicateMessageAttributeNames`.

Reply:
I moved the logic out into the `deduplicateMessageAttributeNames` so that we could unit test it and added some unit tests, let me know what you think!